assets/src/edit-story/components/library/panes/media/deleteDialog.js

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../../../app/api';
+import { Plain } from '../../../../components/button';
+import Dialog from '../../../../components/dialog';
+import { useSnackbar } from '../../../../app/snackbar';
+import { useMedia } from '../../../../app/media';
+
+const DialogBody = styled.div`
+  width: 512px;
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.body1.weight};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const Bold = styled.span`
+  font-weight: bold;
+`;
+
+/**
+ * Display a confirmation dialog for when a user wants to delete a media element.
+ *
+ * @param {Object} props Component props.
+ * @param {number} props.mediaId Selected media element.
+ * @param {boolean} props.showDeleteDialog If dialog should be displayed.
+ * @param {function(boolean)} props.setShowDeleteDialog Callback to toggle dialog display.
+ * @return {null|*} The dialog element.
+ */
+function DeleteDialog({ mediaId, showDeleteDialog, setShowDeleteDialog }) {
+  const {
+    actions: { deleteMedia },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+  const {
+    actions: { deleteMediaElement },
+  } = useMedia();
+
+  const [shouldDelete, setShouldDelete] = useState(false);
+
+  useEffect(() => {
+    const deleteMediaItem = async () => {
+      setShouldDelete(false);
+      setShowDeleteDialog(false);
+      try {
+        await deleteMedia(mediaId);
+        deleteMediaElement({ id: mediaId });
+      } catch (err) {
+        showSnackbar({
+          message: __('Failed to delete media item.', 'web-stories'),
+        });
+      }
+    };
+    shouldDelete ? deleteMediaItem() : null;
+  }, [
+    deleteMedia,
+    deleteMediaElement,
+    mediaId,
+    setShowDeleteDialog,
+    shouldDelete,
+    showSnackbar,
+  ]);
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <Dialog
+      open={showDeleteDialog}
+      onClose={() => setShowDeleteDialog(false)}
+      title={__('Delete image/video?', 'web-stories')}
+      actions={
+        <>
+          <Plain
+            data-testid="cancel"
```

Comment:
Test id should not be necessary here - you can easily get this component by text or role. Test id should only be a last resort, when all else fails. And similarly below for `delete`.

Reply:
Hmm why is that? I'd think that `data-testid` is more a description of the functionality, which is something that is less likely to change than the text chosen.

Reply:
Unit tests are more useful ¹, the more they resemble real user behavior. Real users look for buttons by looking at their text, layout or assistive properties (like labels or roles) - not by looking for an element with a certain DOM data attribute.

See the [guiding principles](https://testing-library.com/docs/guiding-principles) behind testing-library, [their guide to choosing a query](https://testing-library.com/docs/guide-which-query) or the author, Kent C. Dodds', [own guides on the matter](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query).

**In practical terms**: If the button suddenly changes text, users might be confused. It seems fair that your test would too.

Even better than a simple `getByText` (which might also find a headline or a paragraph) is `getByRole('button', { name: /cancel/i })` –  the ["accessible name"](https://www.w3.org/TR/accname-1.1/) here comes from the actual text on the button (with caveats such as `aria-` properties taking precedence if present) and would still break, if the text changes, but would not match other elements, that aren't buttons.

And note that you don't have to manually put a `role="button"` on the element - any `<button />` has the proper implicit role.

¹ For a well-chosen definition of "useful", that is 😉

Reply:
Ah thanks for the guides! Used `getByRole` :)